### DISCUSSION TITLE
My authentication credentials expire every time I quit Submariner

### DISCUSSION THREAD

**Discussion by simensen** [SubmarinerApp/Submariner#134] [Q&A]
Hello,

I'm using "token-based authentication" to connect with Navidrome instances. Whether I am connecting to just one or all four of my instances, restarting Submariner requires me to enter my credentials again. I need to find out if this is by design, a bug, or if there is some setting I need to change on Submariner or Navidrome. 

![image](https://github.com/NattyNarwhal/Submariner/assets/191200/9b67966d-c607-46c8-8490-2f45e11347c5)
![image](https://github.com/NattyNarwhal/Submariner/assets/191200/519202b4-f741-4bd0-8e64-cb9f4a3069db)

----

To be clear, each of the four instances has the same username ("beausimensen") but a different password. Entering the correct password for the server enables me to browse and play music vis Submariner, so I know the credentials are working and are connecting to the right Navidrome instance. It's just that this authentication session seems to vanish the next time I open Submariner.

**Comment by NattyNarwhal** [comment]
I think this might be Keychain related. Do you see your server entries under Keychain Access?

**Reply by simensen** [reply]
I do not see anything under Keychain Access. I tried to authenticate after restarting while watching console. I see the following which looks like it could be sus:

```
default	20:54:25.087409-0500	Submariner	update internet keychain
default	20:54:25.088697-0500	secd	Submariner[95164]/1#6 LF=0 update Error Domain=NSOSStatusErrorDomain Code=-25303 "attribute path not found in class genp" (errKCNoSuchAttr / errSecNoSuchAttr:  / The attribute does not exist.) UserInfo={numberOfErrorsDeep=0, NSDescription=attribute path not found in class genp}
default	20:54:25.135636-0500	Submariner	no previous integrity acl exists; making a new one
default	20:54:25.142709-0500	Submariner	0x6000031f8f00 commited /Users/beausimensen/Library/Keychains/login.keychain-db.sb-15bf8240-pnSJzs to /Users/beausimensen/Library/Keychains/login.keychain-db
```

**Reply by simensen** [reply]
Hm, it might be worth noting I was using the MAS version. I replaced it with a fresh download, and now it seems as though things are being added to my keychain correctly!

**Reply by NattyNarwhal** [reply]
I think you're looking at GH-125, which I did push a release to fix. I don't remember hearing any notification from the MAS though, so it's possible it may be stuck in review limbo 😬
